pkg/egresspolicy/egresspolicy.go

```diff
@@ -0,0 +1,520 @@
+	id podID
+	// ips are pod's unique IPs
+	ips []string
+}
+
+var (
+	allowAllNamespacesRequirement = slim_metav1.LabelSelectorRequirement{
+		Key:      k8sConst.PodNamespaceLabel,
+		Operator: slim_metav1.LabelSelectorOpExists,
+	}
+)
+
+// EgressPolicyConfig is the internal representation of Cilium Egress NAT Policy.
+type EgressPolicyConfig struct {
+	// id is the parsed config name and namespace
+	id k8s.ServiceID
+	// uid is the unique identifier assigned by Kubernetes
+	uid types.UID
+
+	endpointSelectors []api.EndpointSelector
+	dstCIDRs          []*net.IPNet
+	egressIp          net.IP
+}
+
+// policyConfigSelectsPod determines if the given pod is selected by the policy
+// config based on matching labels of config and pod.
+func (config *EgressPolicyConfig) policyConfigSelectsPod(podInfo *podMetadata) bool {
+	labelsToMatch := labels.Set(podInfo.labels)
+	nsLabels := labels.Set(podInfo.nsLabels)
+	for k, v := range nsLabels {
+		labelsToMatch[policy.JoinPath(k8sConst.PodNamespaceMetaLabels, k)] = v
+	}
+	for _, selector := range config.endpointSelectors {
+		if selector.Matches(labelsToMatch) {
+			return true
+		}
+	}
+	return false
+}
+
+func Parse(cenp *v2.CiliumEgressNATPolicy) (*EgressPolicyConfig, error) {
+	name := cenp.ObjectMeta.Name
+	if name == "" {
+		return nil, fmt.Errorf("CiliumEgressNATPolicy must have a name")
+	}
+
+	//TODO implement sanitization for CRD fields
+	var dstCidrList []*net.IPNet
+
+	for _, cidrString := range cenp.Spec.DestinationCIDRs {
+		_, cidr, err := net.ParseCIDR(cidrString)
+		if err != nil {
+			log.WithError(err).Warn("error parsing cidr")
+			continue
+		}
+		dstCidrList = append(dstCidrList, cidr)
+	}
+
+	var endpointSelectorList []api.EndpointSelector
+
+	for _, egressRule := range cenp.Spec.Egress {
+		if egressRule.NamespaceSelector != nil {
+			prefixedNsSelector := egressRule.NamespaceSelector
+			matchLabels := map[string]string{}
+			// We use our own special label prefix for namespace metadata,
+			// thus we need to prefix that prefix to all NamespaceSelector.MatchLabels
+			for k, v := range egressRule.NamespaceSelector.MatchLabels {
+				matchLabels[policy.JoinPath(k8sConst.PodNamespaceMetaLabels, k)] = v
+			}
+			egressRule.NamespaceSelector.MatchLabels = matchLabels
+
+			// We use our own special label prefix for namespace metadata,
+			// thus we need to prefix that prefix to all NamespaceSelector.MatchLabels
+			for i, lsr := range egressRule.NamespaceSelector.MatchExpressions {
+				lsr.Key = policy.JoinPath(k8sConst.PodNamespaceMetaLabels, lsr.Key)
+				egressRule.NamespaceSelector.MatchExpressions[i] = lsr
+			}
+
+			// Empty namespace selector selects all namespaces (i.e., a namespace
+			// label exists).
+			if len(egressRule.NamespaceSelector.MatchLabels) == 0 && len(egressRule.NamespaceSelector.MatchExpressions) == 0 {
+				egressRule.NamespaceSelector.MatchExpressions = []slim_metav1.LabelSelectorRequirement{allowAllNamespacesRequirement}
```

Comment:
Any reason you define `allowAllNamespacesRequirement` as a variable of the package instead of just defining it here?

Reply:
Not really. I moved it to a local variable.